salt/modules/virt.py

```diff
@@ -2500,16 +2501,533 @@ def revert_snapshot(name, vm_snapshot=None, cleanup=False, **kwargs):
+            result['cpu'] = cpu
+
+        elif child.tagName == "power_management":
+            power_management = []
+            for node in child.childNodes:
+                if node.nodeType != xml.dom.Node.ELEMENT_NODE:
+                    continue
+                power_management.append(node.tagName)
+            if power_management:
+                result['power_management'] = power_management
+
+        elif child.tagName == "migration_features":
+            migration = {
+                'live': bool(child.getElementsByTagName('live')),
+                'transports': []
+            }
+            for transport in child.getElementsByTagName('uri_transport'):
+                migration['transports'].append(_get_xml_element_text(transport))
+            result['migration'] = migration
+
+        elif child.tagName == "topology":
+            cells = []
+            for cell in child.getElementsByTagName('cell'):
+                cells.append(_parse_caps_cell(cell))
+            result['topology'] = {'cells': cells}
+
+        elif child.tagName == 'cache':
+            banks = []
+            for bank in child.getElementsByTagName('bank'):
+                banks.append(_parse_caps_bank(bank))
+            result['cache'] = {'banks': banks}
+
+    secmodels = []
+    for secmodel in host.getElementsByTagName('secmodel'):
+        model = {
+            'model': _get_xml_child_text(secmodel, 'model', None),
+            'doi': _get_xml_child_text(secmodel, 'doi', None)
+        }
+        labels = []
+        for label in secmodel.getElementsByTagName('baselabel'):
+            labels.append({
+                'type': label.getAttribute('type'),
+                'label': _get_xml_element_text(label)
+            })
+        if labels:
+            model['baselabels'] = labels
+        secmodels.append(model)
+    if secmodels:
+        result['security'] = secmodels
+
+    return result
+
+
+def capabilities(**kwargs):
+    '''
+    Return the hypervisor connection capabilities.
+
+    ..versionadded:: Fluorine
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' virt.capabilities
+    '''
+    conn = __get_conn(**kwargs)
+    caps = _capabilities(conn)
+    conn.close()
+
+    result = {
+        'host': _parse_caps_host(caps.getElementsByTagName('host')[0]),
+        'guests': []
+    }
+
+    for guest in caps.getElementsByTagName('guest'):
+        result['guests'].append(_parse_caps_guest(guest))
+
+    return result
+
+
+def _parse_caps_supported(node):
+    '''
+    Return a boolean value for a node with a supported attribute.
+    None is returned if there is no 'supported' attribute or if its
+    value is not one of 'yes' or 'no'.
+    '''
+    supported = node.getAttribute('supported')
+    if not supported:
+        return None
+    if supported == 'yes':
+        return True
+    elif supported == 'no':
+        return False
+    return None
```

Comment:
SESE, please? 

Also when you say "return a boolean value" in the description, yet you make it like an old Pentium CPU: a) "true", b) "false" and c) "I don't know" (None). :wink: